enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,279 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports running an operator bundle using OLM via the run bundle command, which is suited for the SQLite-based index format. This enhancement will address how the Operator SDK will support the new File-Based Catalog format for operator catalogs.  More Information on FBCs and how operator bundles are created using Operator SDK can be found at the links below:
+
+- [File-Based Catalogs](https://olm.operatorframework.io/docs/reference/file-based-catalogs/)
+- [File-Based Catalog Issue](<https://github.com/k8s-operatorhub/community-operators/discussions/505>)
+- [Operator-Bundles](<https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md>)
+
+The declarative config is a struct defined by the Operator Registry that represents the index in a format that is easily transformed to a `YAML/JSON` format. The Operator SDK will invoke Operator Registry APIs to generate various components of a declarative config and eventually insert them into the declarative config object. This object will then be converted to a human-readable form like `YAML/JSON`.
+
+```go
+type DeclarativeConfig struct {
+  Packages []Package
+  Channels []Channel
+  Bundles []Bundle
+  Others []Meta
+}
+```
+
+As seen in the struct above, a declarative config has three major components - Bundles, Packages, and Channels. For more information about these subtypes, refer to [Operator Registry DeclarativeConfig definitions](<https://github.com/operator-framework/operator-registry/blob/master/alpha/declcfg/declcfg.go>).
+
+Essentially, a basic File-Based Catalog is composed of these three components in `JSON/YAML` format. The Operator SDK generates each of these components from the Operator Registry APIs in the following manner:
+
+[Bundle](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L239>)
+
+[Package](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L251>)
+
+[Channel](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L260>)
+
+These three structs are used to construct the declarative config object. This declarative config object is a File-Based Catalog representation of an index with a single bundle present in it. This declarative config object can easily be transformed into `YAML/JSON` format.
+
+It is noteworthy that this enhancement proposal does not change the `run bundle` command from a user’s perspective. The command-line arguments and flags will remain the same. However, the Operator SDK will handle adding a newly-generated bundle to an index differently than it currently does. That is the goal of this enhancement proposal.
```

Comment:
I think we wrote this when we were not sure if there would be any user-facing changes. We can remove it, since we already mentioned (in the goals of this EP) that there should be no user facing changes.

Reply:
Done, removed it.